src/Resources/app/administration/src/module/findologic-module/components/findologic-config/index.js

```diff
@@ -1,196 +1,215 @@
+        const case2 = typeof b[prop] === 'string' ? b[prop].toUpperCase() : b[prop]
+
+        let comparison = 0
+        if (case1 > case2) {
+          comparison = order === 'asc' ? 1 : -1
+        } else if (case1 < case2) {
+          comparison = order === 'asc' ? -1 : 1
         }
+        return comparison
+      }
     },
 
-    computed: {
-        /**
-         * @public
-         * @returns {boolean}
-         */
-        showTestButton() {
-            return this.isActive && this.shopkeyAvailable && this.isValidShopkey && this.isStagingShop;
-        },
+    /**
+     * @public
+     */
+    openSalesChannelUrl () {
+      if (this.selectedSalesChannelId !== null) {
+        const criteria = new Criteria()
+        criteria.addFilter(
+          Criteria.equals('id', this.selectedSalesChannelId)
+        )
+        criteria.setLimit(1)
+        criteria.addAssociation('domains')
+        this.salesChannelRepository.search(criteria, Shopware.Context.api).then((searchresult) => {
+          const domain = searchresult.first().domains.first()
+          this._openStagingUrl(domain)
+        })
+      } else {
+        this._openDefaultUrl()
+      }
+    },
 
-        salesChannelRepository() {
-            return this.repositoryFactory.create('sales_channel');
-        }
+    /**
+     * @private
+     */
+    _openDefaultUrl () {
+      const url = `${window.location.origin}?findologic=on`
+      window.open(url, '_blank')
+    },
+
+    /**
+     * @param {Object} domain
+     * @private
+     */
+    _openStagingUrl (domain) {
+      if (domain) {
+        const url = `${domain.url}?findologic=on`
+        window.open(url, '_blank')
+      } else {
+        this._openDefaultUrl()
+      }
+    },
+
+    createdComponent () {
+      this.getCategories()
+    },
+
+    /**
+     * @public
+     */
+    getCategories () {
+      this.isLoading = true
+
+      const translatedCategories = []
+      this.categoryRepository.search(this.criteria, Shopware.Context.api).then((items) => {
+        this.term = null
+        this.total = items.total
+        items.forEach((category) => {
+
+          translatedCategories.push({
+            value: category.id,
+            name: category.name,
+            label: category.translated.breadcrumb.join(' > ')
+          })
+        })
+
+        this.categories = translatedCategories.sort(this.compare('label'))
+      }).finally(() => {
+        this.isLoading = false
+      })
+    }
+  },
+
+  computed: {
+    /**
+     * @public
+     * @returns {boolean}
+     */
+    showTestButton () {
+      return this.isActive && this.shopkeyAvailable && this.isValidShopkey && this.isStagingShop
+    },
 
+    salesChannelRepository () {
+      return this.repositoryFactory.create('sales_channel')
+    },
+
+    categoryRepository () {
+      return this.repositoryFactory.create('category')
+
+    },
+
+    criteria () {
```

Comment:
If this should be a property, please give it an appropriate name. "criteria" isn't very descriptive.